src/Orleans/Streams/Internal/StreamSubscriptionHandleImpl.cs

```diff
@@ -79,14 +86,16 @@ public override Task<StreamSubscriptionHandle<T>> ResumeAsync(IAsyncObserver<T>
                 await NextItem(itemTuple.Item1, itemTuple.Item2);
             }
 
-            expectedToken = StreamHandshakeToken.CreateDeliveyToken(batch.SequenceToken);
-
+            if (isRewindable)
+            {
+                expectedToken = StreamHandshakeToken.CreateDeliveyToken(batch.SequenceToken);
+            }
             return null;
         }
 
         public async Task<StreamHandshakeToken> DeliverItem(object item, StreamSequenceToken currentToken, StreamHandshakeToken handshakeToken)
         {
-            if (expectedToken != null)
+            if (isRewindable && expectedToken != null)
```

Comment:
>The subscription handle still retains the expected token, and when the agent is moved, it is this token that becomes the new starting point for delivery, during the handshake process

The expected token will be null, triggering this line:
https://github.com/dotnet/orleans/blob/master/src/OrleansRuntime/Streams/PersistentStream/PersistentStreamPullingAgent.cs#L256 
"if the consumer did not ask for a specific token"
I remember writing this comment.